ansible_collections/arista/avd/plugins/plugin_utils/eos_validate_state_utils/catalog.py

```diff
@@ -0,0 +1,104 @@
+from __future__ import annotations
+
+from functools import cached_property
+
+from ansible.errors import AnsibleError
```

Comment:
Please avoid using AnsibleError here. You can catch the errors in the action plugin and reraise as AnsibleActionError.